utils/imds.go

```diff
@@ -11,23 +12,26 @@ const (
 	CIDRKey           = "vpc-cidr"
 )
 
-func IMDSFetchIPv4CIDR(imdsServerAddr string) (string, error) {
+func IMDSFetchIPv4CIDR(imdsServerAddr string) ([]string, error) {
 	// This method needs an instance mac address to get the VPC ipv4 CIDR.
 	var mac string
 	var err error
 	endpoint := fmt.Sprintf("%s/latest/meta-data/mac", imdsServerAddr)
 	mac, err = HTTPGetAndParseBody(endpoint)
 	if err != nil {
-		return "", fmt.Errorf("failed to determine mac address of instance: %v", err)
+		return nil, fmt.Errorf("failed to determine mac address of instance: %v", err)
 	}
 
 	var cidr string
-	endpoint = fmt.Sprintf("%s/latest/meta-data/network/interfaces/macs/%s/vpc-ipv4-cidr-block",
+	endpoint = fmt.Sprintf("%s/latest/meta-data/network/interfaces/macs/%s/vpc-ipv4-cidr-blocks",
 		imdsServerAddr, mac)
 	cidr, err = HTTPGetAndParseBody(endpoint)
 	if err != nil {
-		return "", fmt.Errorf("Could not get VPC CIDR using mac address %q: %v", mac, err)
+		return nil, fmt.Errorf("Could not get VPC CIDR using mac address %q: %v", mac, err)
 	}
 
-	return cidr, nil
+	var cidrs []string
+	cidrs = append(cidrs, strings.Split(strings.TrimRight(cidr, "\n"), "\n")...)
+
+	return cidrs, nil
```

Comment:
- I believe handling this in `HTTPGetAndParseBody` seems more reasonable 🤔 
- Then you could just replace with
```
cidrs, err := HTTPGetAndParseBody
// err check
return cidrs, nil
```

Reply:
Do you think it is reasonable to split strings in a method called HTTPGetAndParseBody ?

Reply:
I believe the `HTTPGetAndParseBody` function should get the response body, and we should parse it according to the use case. It can be a string, JSON object, or anything else. So I feel it is better to split it in the function from where it is called.

But I agree that the function name should be something more generic such as `HTTPGetResponseBody`.

Let me know if you still feel that we should split it in the `HTTPGetAndParseBody` function.

Reply:
> It can be a string, JSON object, or anything else
- Agreed! so caller should do the transformation as I can now see another place where this is being used

all good!